core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -118,31 +128,33 @@ public GeyserServer(GeyserImpl geyser, int threadCount) {
     }
 
     public CompletableFuture<Void> bind(InetSocketAddress address) {
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        this.bootstrapFuture = this.bootstrap.bind(address).addListener(bindResult -> {
-            if (bindResult.cause() != null) {
-                future.completeExceptionally(bindResult.cause());
-                return;
-            }
-            future.complete(null);
-        });
+        bootstrapFutures = new ChannelFuture[listenCount];
+        for (int i = 0; i < listenCount; i++) {
+            bootstrapFutures[i] = Optional.of(bootstrap)
+                    .map(f -> f.bind(address))
+                    .map(f -> addHandlers(f, this))
+                    .orElseThrow();
+        }
 
-        Channel channel = this.bootstrapFuture.channel();
+        return Bootstraps.allOf(bootstrapFutures);
+    }
 
+    private static ChannelFuture addHandlers(ChannelFuture f, GeyserServer self) {
+        Channel channel = f.channel();
         // Add our ping handler
         channel.pipeline()
-                .addFirst(RakConnectionRequestHandler.NAME, new RakConnectionRequestHandler(this))
-                .addAfter(RakServerOfflineHandler.NAME, RakPingHandler.NAME, new RakPingHandler(this));
-
-        if (this.geyser.getConfig().getBedrock().isEnableProxyProtocol()) {
+                .addFirst(RakConnectionRequestHandler.NAME, new RakConnectionRequestHandler(self))
+                .addAfter(RakServerOfflineHandler.NAME, RakPingHandler.NAME, new RakPingHandler(self));
+        // Add proxy handler
+        if (self.geyser.getConfig().getBedrock().isEnableProxyProtocol()) {
             channel.pipeline().addFirst("proxy-protocol-decoder", new ProxyServerHandler());
         }
-
-        return future;
+        return f;
     }
 
     public void shutdown() {
         try {
+            this.childGroup.shutdownGracefully();
```

Comment:
we should probably be using the same shutdown periods as the others - otherwise add a comment